src/rgw/rgw_lc.cc

```diff
@@ -91,12 +91,7 @@ bool RGWLifecycleConfiguration::_add_rule(LCRule *rule)
     op.obj_tags = rule->get_filter().get_tags();
   }
 
-  /* prefix is optional, update prefix map only if prefix...exists */
-  if (!prefix.empty()) {
-    auto ret = prefix_map.emplace(std::move(prefix), std::move(op));
-    return ret.second;
-  }
-
+  prefix_map.emplace(std::move(prefix), std::move(op));
   return true;
```

Comment:
in case of empty prefixes what happens here, we pass down the empty prefix to list bucket op?

Reply:
yes, sure?

Reply:
I'm grabbing food, will be back in a sec :)

Reply:
Maybe we could then add the check before passing it to the list_objects in the downward `bucket_lc_process` layer, I assume down the cls, there will be one compare call per item in the list to match a prefix, we could avoid that if we don't pass the prefix in the list_objects call?

Reply:
nvm, ignore the comment, looks fine